### PR TITLE
boardsource/3x4 data driven

### PR DESCRIPTION
**Pull request by waffle87**
## Description

As title

## Types of Changes

- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardsource/3x4/info.json [hunk 1]
```diff
@@ -1,27 +1,41 @@
...
-                { "label": "K09", "x": 0, "y": 2 },
-                { "label": "K10", "x": 1, "y": 2 },
-                { "label": "K11", "x": 2, "y": 2 },
-                { "label": "K12", "x": 3, "y": 2 }
-            ]
-        }
+  "manufacturer": "Boardsource",
+  "keyboard_name": "3x4",
+  "maintainer": "waffle87",
+  "bootloader": "caterina",
```
Comment (zvecr) [inline keyboards/boardsource/3x4/info.json:5]:
```suggestion
  "development_board": "promicro",
```

File: keyboards/boardsource/3x4/info.json [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+  "features": {
+    "bootmagic": true,
+    "extrakey": true,
+    "mousekey": true
+  },
+  "matrix_pins": {
+    "cols": ["B6", "B2", "B3", "B1"],
+    "rows": ["F7", "F6", "F5"]
+  },
+  "processor": "atmega32u4",
```
Comment (zvecr) [inline keyboards/boardsource/3x4/info.json:16]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
`development_board` presets various config like bootloader and processor.

In the future it will become a requirement for converters to work.

File: keyboards/boardsource/3x4/keymaps/via/keymap.c [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
+  [2] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+  ),
+  [3] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+  )
```
Comment (zvecr) [inline keyboards/boardsource/3x4/keymaps/via/keymap.c:25]:
```suggestion
  [2] = LAYOUT(
    _______, _______, _______, _______,
    _______, _______, _______, _______,
    _______, _______, _______, _______
  ),
  [3] = LAYOUT(
    _______, _______, _______, _______,
    _______, _______, _______, _______,
    _______, _______, _______, _______
  )
```

File: keyboards/boardsource/3x4/rules.mk [hunk 4]
```diff
@@ -1,17 +0,0 @@
-# MCU name
```
Comment (zvecr) [inline keyboards/boardsource/3x4/rules.mk:1]: Can we align this to the template file `data/templates/keyboard/rules.mk`
```makefile
# This file intentionally left blank
```

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**